modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1373 @@
+        });
+    }
+
+    @Test(dataProvider = "dataForGetException")
+    public void testGetUserBadRequestException(String name, String attributes, String excludeAttributes)
+            throws CharonException, BadRequestException, NotFoundException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema)
+                        CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn("https://localhost:9443/scim2/Me");
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(BadRequestException.class)))
+                .thenThrow(BadRequestException.class);
+        when(userManager.getMe(name, requiredAttributes)).thenThrow(BadRequestException.class);
+
+        //Assertions
+        Assert.assertThrows(BadRequestException.class, () -> {
+            meResourceManager.get(name, userManager, attributes, excludeAttributes);
+        });
+    }
+
+    @DataProvider(name = "dataForTestCreateSuccess")
+    public Object[][] dataToTestCreateSuccess() throws BadRequestException, CharonException, InternalErrorException {
+
+        String scimObjectString = getNewUserSCIMObjectString();
+        User user = getNewUser();
+
+        return new Object[][]{
+                {scimObjectString, "userName", null, user, 201}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestCreateSuccess")
+    public void testCreateSuccess(String scimObjectString, String attributes, String excludeAttributes,
+                                  Object objectUser, int expectedScimResponseStatus)
+            throws BadRequestException, NotFoundException, CharonException, ConflictException {
+
+        User user = (User) objectUser;
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn("https://localhost:9443/scim2/Me");
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+
+        when(userManager.createMe(anyObject(), anyObject())).thenReturn(user);
+
+        SCIMResponse outputScimResponse = meResourceManager.create(scimObjectString, userManager,
+                attributes, excludeAttributes);
+
+        //Assertions
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
```

Comment:
Assert if the expected location header with the URI of the User object is present as a response header

Reply:
Fixed in commit f8e4cf17c52ce61a257e2d77331fff85c69e7f66